core/src/main/java/com/opentable/logging/HttpLogFields.java

```diff
@@ -26,16 +28,36 @@
     @JsonProperty("url")
     String getUrl();
 
+    @JsonProperty("url-hostname")
+    default String getUrlHostname() {
+        return URI.create(getUrl()).getHost();
+    }
+
+    @JsonProperty("url-port")
+    default int getUrlPort() {
+        return URI.create(getUrl()).getPort();
+    }
+
+    @JsonProperty("url-pathname")
+    default String getUrlPath() {
+        return URI.create(getUrl()).getPath();
+    }
+
+    @JsonProperty("url-querystring")
+    default String getUrlQuery() {
+        return URI.create(getUrl()).getQuery();
+    }
+
     @JsonProperty("status")
     int getStatus();
 
     @JsonProperty("duration")
     long getDurationMicros();
 
-    @JsonProperty("bodysize")
-    Long getBodySize();
+    @JsonProperty("body-size")
+    long getBodySize();
```

Comment:
Did you look at implementers of this?  Should we now expect NPEs?  Strictly speaking, this is a breaking API change.